File: HomeCareApp-React-main (2)/HomeCareApp-React-main/api/Models/Appointment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HomeCareApp.Models
{
    public class Appointment
    {
        public int Id { get; set; }

        [Required]
        [Display(Name = "Client")]
        public string ClientId { get; set; } = string.Empty;

        [ForeignKey(nameof(ClientId))]
        public User? Client { get; set; }

        [Required]
        [Display(Name = "Availability")]
        public int AvailabilityId { get; set; }

        [ForeignKey(nameof(AvailabilityId))]
        public Availability? Availability { get; set; }

        [Required(ErrorMessage = "Task description is required.")]
        [StringLength(200, ErrorMessage = "Task description cannot be longer than 200 characters.")]
        public string TaskDescription { get; set; } = string.Empty;

        [Required]
        [DataType(DataType.Time)]
        public TimeSpan StartTime { get; set; }

        [Required]
        [DataType(DataType.Time)]
        public TimeSpan EndTime { get; set; }

        [Required]
        [RegularExpression(@"^(Booked|Completed|Cancelled)$", ErrorMessage = "Invalid status value.")]
        public string Status { get; set; } = "Booked";
    }
}

File: HomeCareApp-React-main (2)/HomeCareApp-React-main/frontend/src/types/Appointment.ts
export interface Appointment {
  id: number;
  availabilityId: number;
  clientId?: string;
  clientName?: string;
  personnelId?: string;
  personnelName?: string;
  date: string;
  startTime: string;
  endTime: string;
  taskDescription: string;
  status: "Booked" | "Completed" | "Cancelled";
}

export interface AppointmentCreatePayload {
  availabilityId: number;
  clientId?: string;
  taskDescription: string;
  startTime: string;
  endTime: string;
  status: string;
}
